packages/adapter-ccxt/src/CCXTAdapter.ts

```diff
@@ -82,17 +92,28 @@ export class CCXTAdapter implements IAdapter<ExecutionResult, CCXTStep> {
    }
 
    private async initializeExchanges(): Promise<void> {
-      for (const [exchangeId, exchangeConfig] of Object.entries(
-         this.adapterConfig.exchanges
-      ) as Array<[ExchangeId, ApiCredentials]>) {
-         const exchangeService = this.exchangeFactory.createExchangeService(
-            exchangeId,
-            exchangeConfig
-         );
-         this.exchangeServiceRepo.setExchangeService(
-            exchangeId,
-            exchangeService
-         );
-      }
+      const initializationPromises = this.adapterConfig.exchanges.map(
+         async (exchangeConfig) => {
```

Comment:
doesn't this whole block need to return something in order for you to do `await Promise.all(initializationPromises);` after?

Reply:
It didn't need to be async so refactored to remove that.